.github/workflows/build.yml

```diff
@@ -0,0 +1,73 @@
+name: Build and test
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  create:
+
+jobs:
+  build-and-test-code:
+    runs-on: ubuntu-20.04
+    steps:
+      # Set fetch-depth: 0 to fetch all history for all branches and tags.
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+      
+      - name: create checksum file
+        run: find . -type f -name "*.gradle*" -o -name "gradle-wrapper*" | xargs shasum | sort > checksum.txt && cat checksum.txt
+
+      - name: download dependencies
+        run: ./gradlew downloadDependencies
```

Comment:
Could we create a new action to mirror the behavior we have in circleci - that is, defaulting the flags we want to run with:
```
gradle:
    description: 'Run the provided gradle command'
    parameters:
      args:
        type: string
      when:
        default: "on_success"
        type: enum
        enum: ["on_fail", "on_success", "always"]
    steps:
      - run:
          name: << parameters.args >>
          command: ./gradlew << parameters.args >> --info --max-workers=2 -Dorg.gradle.jvmargs=-Xmx2g -Dorg.gradle.console=plain --continue
          when: << parameters.when >>
```

Reply:
Okay. let me explore that. but is it possible to hold on this for a while?

Reply:
Sure, but I'd encourage doing it before rolling GHA out to other repos so we can minimize duplicated code.